src/conditions/mod.rs

```diff
@@ -34,3 +37,66 @@ dyn_clone::clone_trait_object!(ConditionConfig);
 pub type ConditionDescription = ComponentDescription<Box<dyn ConditionConfig>>;
 
 inventory::collect!(ConditionDescription);
+
+#[derive(Debug, Deserialize, Serialize, Clone)]
+#[serde(untagged)]
+pub enum AnyCondition {
+    String(String),
+    Map(Box<dyn ConditionConfig>),
+}
+
+impl AnyCondition {
+    pub fn build(&self) -> crate::Result<Box<dyn Condition>> {
+        match self {
+            AnyCondition::String(s) => RemapConfig { source: s.clone() }.build(),
+            AnyCondition::Map(m) => m.build(),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[derive(Deserialize, Debug)]
+    struct Test {
+        condition: AnyCondition,
+    }
+
+    #[test]
+    fn deserialize_anycondition_default() {
+        let conf: Test = toml::from_str(r#"condition = ".nork == false""#).unwrap();
+        assert_eq!(
+            "Test { condition: String(\".nork == false\") }",
+            format!("{:?}", conf)
+        )
```

Comment:
I'd rather test these as:

```suggestion
        assert_eq!("String(\".nork == false\")", format!("{:?}", conf.condition)
```

to strip the parts that are only relevant to the test setup, and not the thing to be asserted.

Additionally, a raw string in the assertion will make the quoted string inside clearer:


```suggestion
        assert_eq!(r#"String(".nork == false")"#, format!("{:?}", conf.condition)
```